.github/actions/setup/action.yml

```diff
@@ -20,38 +24,42 @@ inputs:
 runs:
   using: "composite"
   steps:
+    - name: Install poetry
+      shell: bash
+      run: |
+        pipx install poetry
+        which poetry
+        poetry --version  # Check poetry installation
+    - name: Lock poetry  # lock file is needed for the next step, but we don't track our lock file
+      shell: bash
+      run: poetry lock
```

Comment:
@mikapfl @lewisjared this is a slightly interesting learning. Basically, if you don't track the lock file in the repo, you need to make it in CI (somewhere, with this workflow it is an explicit step). 

The question is what implications this has for the copier repo. My instinct is that we go the very simple approach i.e. we just add an extra question to our copier repo, specifically: "Do you want to track your poetry.lock file?" If yes, we add this line to CI and ignore the lock file, if no, we don't add this line to CI and track the lock file.

The other option would be to split the copier repo into two: one for repos which track the lock file (which are more application like), one for those that don't (libraries). In my head that initially made sense, but it feels like massive overkill here to me. I think, having also seen how some of the application repositories are coming together, I don't think you would often (if ever) reach for the 'core copier python repo' if you were actually doing a full application (reason being, usually you want something specific e.g. give me a repo that has a notebook-based workflow setup, or something super general, e.g. https://gitlab.com/znicholls/copier-basic-python-repository). That's more of a rant than I wanted, but hopefully the point is clear: you could split the 'core copier python repo' but I don't think it actually solves any problems.

Reply:
@mikapfl @lewisjared I'm going to merge but if you have any thoughts, fire away here. If I don't hear anything, I'll just go with my first instinct.